src/triggers.js

```diff
@@ -337,7 +337,16 @@ export function getResponseObject(request, resolve, reject) {
       }
       response = {};
       if (request.triggerName === Types.beforeSave) {
-        response['object'] = request.object._getSaveJSON();
+        try {
+          response['object'] = request.object._getSaveJSON();
+        } catch (error) {
+          // https://github.com/parse-community/parse-server/issues/7192
+          const e = resolveError(error, {
+            code: Parse.Error.VALIDATION_ERROR,
+            message: 'Script success. But return object failed to converted to valid JSON.',
```

Comment:
What does `script success` mean?

Reply:
It means the beforeSave script actually executed without throw. I copied this from another location in the code. It makes sense to me because we are failing not because of the beforeSave throw exception but the output of the beforeSave is not valid. Of course in our test the input was not valid even before beforeSave, but it also could be that the beforeSave messed up the iso, there is no way for the code here to know where exactly this happened.

Reply:
This should be more specific, "script success" is too vague. Never assume that existing code is flawless when you copy/paste it into your PR.

Reply:
Do you think "Even though the beforeSave script executed successfully without throwing exception, the returned object could not be converted to valid JSON." makes sense? I do not think I can assume it is failing because of Date format conversion even though that's how we found the problem.

Reply:
> I'm not why it should mention what succeeded, instead of just saying what failed.  How about `afterSave failed because object could not be converted to JSON.`? I suggest to also change the other code from where you copied the message, if that makes sense.

Reply:
I changed to what I posted above.